graspologic/preprocessing/graph_cuts.py

```diff
@@ -0,0 +1,383 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import logging
+import random
+from typing import Any, Callable, List, NamedTuple, Optional, Tuple, Union
+import networkx as nx
+import numpy as np
+
+
+LARGER_THAN_INCLUSIVE = "LARGER_THAN_INCLUSIVE"
+LARGER_THAN_EXCLUSIVE = "LARGER_THAN_EXCLUSIVE"
+SMALLER_THAN_INCLUSIVE = "SMALLER_THAN_INCLUSIVE"
+SMALLER_THAN_EXCLUSIVE = "SMALLER_THAN_EXCLUSIVE"
+
+
+def _filter_function_for_make_cuts(
+    cut_threshold: Union[int, float],
+    cut_process: str,
+) -> Callable[[Tuple[Any, Union[int, float]]], bool]:
+    filter_functions = {
+        LARGER_THAN_EXCLUSIVE: lambda x: x[1] > cut_threshold,
+        LARGER_THAN_INCLUSIVE: lambda x: x[1] >= cut_threshold,
+        SMALLER_THAN_EXCLUSIVE: lambda x: x[1] < cut_threshold,
+        SMALLER_THAN_INCLUSIVE: lambda x: x[1] <= cut_threshold,
+    }
+    if cut_process not in filter_functions:
+        raise ValueError(f"Provided cut_process '{cut_process}' is not a valid value")
+    return filter_functions[cut_process]
+
+
+class DefinedHistogram(NamedTuple):
+    """
+    Contains the histogram and the edges of the bins in the histogram.
+    The bin_edges will have a length 1 greater than the histogram, as it defines the minimal and maximal edges as well
+    as each edge in between.
+    """
+
+    histogram: np.ndarray
+    bin_edges: np.ndarray
+
+
+def histogram_edge_weight(
+    graph: nx.Graph,
+    bin_directive: Union[int, List[Union[float, int]], np.ndarray, str] = 10,
+    weight_attribute: str = "weight",
+) -> DefinedHistogram:
+    """
+    Generates a histogram of the edge weights of the provided graph. Histogram function is fundamentally proxied
+        through to numpy's `histogram` function, and bin selection follows `numpy.histogram` processes.
+
+    Parameters
+    ----------
+
+    graph : nx.Graph
+        The graph. No changes will be made to it.
+    bin_directive : Union[int, List[Union[float, int]], numpy.ndarray, str]
+        Is passed directly through to numpy's "histogram" (and thus, "histogram_bin_edges") functions.
+        See: https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.histogram_bin_edges.html#numpy.histogram_bin_edges
```

Comment:
have been trying to use intersphinx for links like this because it's more stable to changes in doc locations

Reply:
I legit have no idea what that even means.  What's the cliff notes on it?

Reply:
can reference documentation in other packages via **`programatically`**
ex: https://github.com/microsoft/graspologic/blob/c3ea3d98de09d6d3a79840f0472b3889ada71d2f/graspologic/embed/ase.py#L46

Reply:
which then renders nicely AND hyperlinks https://graspologic.readthedocs.io/en/latest/reference/embed.html#single-graph-embedding

Reply:
super nice IMO, won't break as much when location of links changes, and by default will go to stable release docs

Reply:
I got it working for numpy, but it is just not working for networkx.  I don't know what I'm missing but I hate this so bad right now

Reply:
I think I know why! 1 sec